### PR TITLE
Fix Firefox 59

### PR DESCRIPTION
**Pull request by kekkokk**
Ok so,
With Firefox 59 (now in beta) there are lots of problems.

1) It generates offer with m line video before audio. this caused an "error" when generating the answer.
For more information see issue: https://github.com/lynckia/licode/issues/1141

2) Apparently when creating offer, even passing offertoreceiveaudio: false and offertoreceivevideo: false, it creates offer with 2 media in sendrecv. 
I don't know if it's a bug or not but you can follow the discussion from comment 3 [HERE (Bugzilla)](https://bugzilla.mozilla.org/show_bug.cgi?id=1433953)
you can try the example I posted on comment 3 with Firefox 59.
This caused problem with firefox simulcast publisher because the offer generated was audio in sendonly and video in sendrecv. we were generating an answer with both medias in recvonly (the first stream parsed) and when setting answer Firefox throwed an error.
Since licode can't handle multiple direction in one peerconnection my solution was to munge sdp object and set sendonly and recv only in the appropriate places.

Everything's seems to be working fine now, I haven't done any class test so maybe I need help with that.
Also obviously thanks @Trisfald  who helped me with c++ part

ps. there is also a new variable getSessionVersion that we use to increment at every ice restart. maybe you can keep it so when multiple sdp negotiation will be supported, you will increment that.

### CODE REVIEW

File: erizo/src/erizo/SdpInfo.h [hunk 1]
```diff
@@ -254,6 +258,11 @@ class SdpInfo {
   */
   const std::vector<Rid>& rids() const { return rids_; }
 
+  int getSessionVersion() const;
```
Comment (jcague) [inline erizo/src/erizo/SdpInfo.h]: I think it's always good to add support to SDP fields, but I'd do that in separate PRs to reduce the scope of fixes.

File: erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js [hunk 2]
```diff
@@ -92,6 +92,7 @@ const BaseStack = (specInput) => {
       localDesc.sdp = that.enableSimulcast(localDesc.sdp);
     }
     localSdp = SemanticSdp.SDPInfo.processString(localDesc.sdp);
+    SdpHelpers.forceDirection(localSdp, isSubscribe);
```
Comment (jcague) [inline erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js]: I think this code should only be added to the Firefox stack.

File: erizo_controller/erizoJS/models/SessionDescription.js [hunk 3]
```diff
@@ -205,23 +205,34 @@ class SessionDescription {
...
       ipVer: 4,
       address: '127.0.0.1' });
     sdp.name = 'LicodeMCU';
 
     sdp.msidSemantic = { semantic: 'WMS', token: generateRandom(10) };
 
-    if (info.hasAudio()) {
-      const media = getMediaInfoFromDescription(info, sdp, 'audio');
-      sdp.addMedia(media);
+    if (info.getFirstMediaReceived() === 'audio') {
```
Comment (jcague) [inline erizo_controller/erizoJS/models/SessionDescription.js]: I see the problem, but I prefer a less hacky solution... would it be possible to add medias in the same order they appear in the bundle group field?

Comment (kekkokk) [reply erizo_controller/erizoJS/models/SessionDescription.js]: Yeah seems a little hacky way also to me but unfortunately the datas are fetched from SdpInfo as separate field (audio and video).
Until there's a refactor in Sdpinfo.cpp that support an ordered array of media (that I tought will be done in the future with the singlepeerconnection feature) I can't think other ways to pass the order than this hacky variable.

File: erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js (hunk 2 above)
Comment (kekkokk) [reply erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js]: Ok, I don't think could harm to force here too but if you prefer I will move to Firefox Stack

File: erizo/src/erizo/SdpInfo.h (hunk 1 above)
Comment (kekkokk) [reply erizo/src/erizo/SdpInfo.h]: ok I'll move out

File: erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js (hunk 2 above)
Comment (kekkokk) [reply erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js]: Also they replied to me that Firefox now support transceivers, so we have to set direction in videoTransceiver and audioTransceiver. I think it's more an adapter job since they already do this in safari. So this will be moved in firefox stack as a temporary fix until adapter came out with their own solution

**Comment by kekkokk** [comment]
Also read this:
https://github.com/webrtc/adapter/issues/760#issuecomment-361874732

Maybe they'll fix in 59 before the realease.

Still the media order is a todo fix for us

**Comment by kekkokk** [comment]
I made two separates PR. close this
